cmake/git.cmake

```diff
@@ -0,0 +1,19 @@
+# Use git if it is installed, to set BUILD_VERSION
+
+# https://cmake.org/cmake/help/latest/module/FindGit.html
+find_package(Git QUIET)
+if(GIT_FOUND)
+  execute_process(COMMAND ${GIT_EXECUTABLE} describe
+    WORKING_DIRECTORY                ${ZEPHYR_BASE}
+    OUTPUT_VARIABLE                  BUILD_VERSION
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    ERROR_STRIP_TRAILING_WHITESPACE
+    ERROR_VARIABLE                   stderr
+    RESULT_VARIABLE                  return_code
+  )
+  if(return_code)
+    message(STATUS "git describe failed: ${stderr}; ${KERNEL_VERSION_STRING} will be used instead")
```

Comment:
'KERNEL_VERSION_STRING'

is set by version.cmake. This patch moves git.cmake before version.cmake, making this variable unavailable.

Could you also check other variables? (or retain the location)

Reply:
Correct. I looked for other use-before-def, but could only find that.
I've moved inclusion of version.cmake up to be earlier.